gettsim/typing.py

```diff
@@ -17,17 +19,78 @@ def check_if_series_has_internal_type(series, internal_type):
 
     Returns
     -------
-    out : bool
-        Return check variable.
+    out : adjusted pd.Series
     """
+
+    # Conversion to float
     if internal_type == float:
-        out = is_float_dtype(series) or is_integer_dtype(series)
-    elif internal_type == bool:
-        out = is_bool_dtype(series)
+        # Conversion from boolean to float fails
+        if is_bool_dtype(series):
+            raise ValueError("Conversion of boolean to float is not supported.")
+        elif not is_float_dtype(series):
+            try:
+                series = series.astype(float)
+            except ValueError:
+                raise ValueError(f"Conversion of data type to {internal_type} failed.")
```

Comment:
Actually, we should collect all errors at a higher level and list them together in a table rather than raising them one by one. From a user perspective, that is really annoying (especially since I don't expect many of our eventual users to worry a lot about dtypes).

Just add a try / except block in the calling code. Or is there a smarter way to do this, @janosg ?